src/renderer/redux/selectors/subscriptions.js

```diff
@@ -3,24 +3,130 @@ import {
   selectAllClaimsByChannel,
   selectClaimsById,
   selectAllFetchingChannelClaims,
-  makeSelectClaimForUri,
+  makeSelectChannelForClaimUri,
+  selectClaimsByUri,
+  parseURI,
 } from 'lbry-redux';
 
-// get the entire subscriptions state
+// Returns the entire subscriptions state
 const selectState = state => state.subscriptions || {};
 
+// Returns the list of channel uris a user is subscribed to
+export const selectSubscriptions = createSelector(selectState, state => state.subscriptions);
+
+// Fetching list of users subscriptions
 export const selectIsFetchingSubscriptions = createSelector(selectState, state => state.loading);
 
-export const selectNotifications = createSelector(selectState, state => state.notifications);
+// The current view mode on the subscriptions page
+export const selectViewMode = createSelector(selectState, state => state.viewMode);
 
-// list of saved channel names and uris
-export const selectSubscriptions = createSelector(selectState, state => state.subscriptions);
+// Fetching any claims that are a part of a users subscriptions
+export const selectSubscriptionsBeingFetched = createSelector(
+  selectSubscriptions,
+  selectAllFetchingChannelClaims,
+  (subscriptions, fetchingChannelClaims) => {
+    const fetchingSubscriptionMap = {};
+    subscriptions.forEach(sub => {
+      const isFetching = fetchingChannelClaims && fetchingChannelClaims[sub.uri];
+      if (isFetching) {
+        fetchingSubscriptionMap[sub.uri] = 1;
+      }
+    });
+
+    return fetchingSubscriptionMap;
+  }
+);
+
+export const selectUnreadByChannel = createSelector(selectState, state => state.unread);
+
+// Returns the current total of unread subscriptions
+export const selectUnreadAmount = createSelector(selectUnreadByChannel, unreadByChannel => {
+  let badges = 0;
+return 0;
+  if (!Object.keys(unreadByChannel).length) {
```

Comment:
Store `Object.keys(unreadByChannel)` in a local variable, you are using it multiple times